src/mesh/voxel_cloud.rs

```diff
@@ -228,6 +239,163 @@ impl VoxelCloud {
         }
     }
 
+    /// Computes boolean intersection (logical AND operation) of the current and
+    /// another Voxel cloud. The Voxel clouds have to intersect and the voxel
+    /// dimension must be equal. The current Voxel cloud will be mutated and
+    /// resized to the size and position of an intersection of the two Voxel
+    /// clouds (not the contained volumes).
+    pub fn boolean_intersection_mut(&mut self, other: &VoxelCloud) {
+        assert!(
+            approx::relative_eq!(self.voxel_dimensions, other.voxel_dimensions()),
+            "The two voxel clouds don't contain voxels of the same size"
+        );
```

Comment:
i can imagine this would work also for voxel clouds with voxels of different sizes - by requesting the other through cartesian coordinates of self. why not, it just seems a bit dirty.

Reply:
I am not absolutely sure what the correct behavior should be, but one more point towards using cartesian coords instead is that if we ever make `VoxelCloud` available as a variable type, the user will be able to create all kinds of voxel clouds and won't be able to not trigger this assert.

Reply:
let's do it with the cartesian coords. it is more general, allows for nice voxel operations and doesn't cost much